src/BlockTemplatesController.php

```diff
@@ -855,12 +855,36 @@ public function update_template_permalink( $value, $option ) {
 	 * @return string THe actual permalink assigned to the page. May differ from $permalink if it was already taken.
 	 */
 	protected function sync_endpoint_with_page( $page, $page_slug, $permalink ) {
-		$matching_page = get_page_by_path( $permalink );
+		$matching_page = get_page_by_path( $permalink, OBJECT, 'page' );
+
+		/**
+		 * Filters whether to attempt to guess a redirect URL for a 404 request.
+		 *
+		 * Returning a false value from the filter will disable the URL guessing
+		 * and return early without performing a redirect.
+		 *
+		 * @since 5.5.0
+		 *
+		 * @param bool $do_redirect_guess Whether to attempt to guess a redirect URL
+		 *                                for a 404 request. Default true.
+		 */
+		if ( ! $matching_page instanceof WP_Post && apply_filters( 'do_redirect_guess_404_permalink', true ) ) {
+			// If it is a subpage and url guessing is on, then we will need to get it via postname as path will not match.
+			$query = new \WP_Query(
+				[
+					'post_type' => 'page',
+					'name'      => $permalink,
+				]
+			);
+
+			$matching_page = $query->have_posts() ? $query->posts[0] : null;
+		}
 
 		if ( $matching_page && 'publish' === $matching_page->post_status ) {
 			// Existing page matches given permalink; use its ID.
 			update_option( 'woocommerce_' . $page_slug . '_page_id', $matching_page->ID );
```

Comment:
```suggestion
		// Existing page matches given permalink; use its ID.
		if ( $matching_page && 'publish' === $matching_page->post_status ) {
			update_option( 'woocommerce_' . $page_slug . '_page_id', $matching_page->ID );
```

I know that you did not touch this section, but I noticed that other checks have the comment before the check. What are your thoughts on moving this comment out of the check so that it's consistent with the other comments?

Reply:
It states "matches" which implies the condition's verified and makes sense to be inside. The other one describes the if/else hence it being positioned outside.